experimental/web/sample_dynamic/build_sample.sh

```diff
@@ -14,8 +14,29 @@
 #
 # Usage:
 #   build_sample.sh (optional install path) && serve_sample.sh
+#
+# The desired host install directory can be passed as the first argument.
+# Otherwise, it looks for an install directory under path set in the environment
+# variable IREE_HOST_BUILD_DIR (default build-host). The build directory for the
+# empscripten build is taken from the environment variable
+# IREE_EMPSCRIPITEN_BUILD_DIR, defaulting to "build-empscripten". Designed for
+# CI, but can be run manually. It reuses the build directory if it already
+# exists.
+#
+# NOTE: This is different from most of build scripts we use for CI because it is
+# intended to also be runnable by humans with minimal configuration.
+
+set -euo pipefail
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+
+HOST_BUILD_DIR="${IREE_HOST_BUILD_DIR:-${ROOT_DIR}/build-host}"
+BUILD_DIR="${IREE_EMPSCRIPITEN_BUILD_DIR:-build-emscripten}"
+INSTALL_ROOT="${1:-${HOST_BUILD_DIR}/install}"
+CMAKE_BIN=${CMAKE_BIN:-$(which cmake)}
+SOURCE_DIR=${ROOT_DIR}/experimental/web/sample_dynamic
+BINARY_DIR=${BUILD_DIR}/experimental/web/sample_dynamic
```

Comment:
Any chance to use `source build_tools/cmake/setup_build.sh` here?

Reply:
Possibly... though these scripts interact with the build directory more closely than other scripts do

Reply:
I looked at using that, but ended up not because this script has other requirements about being runnable manually, which mostly consrains assuming we're in the root directory to start.

Reply:
I looked at using that and then decided not to because this script has other requirements about being runnable manually, which mostly means not assuming we're in the root directory to start. But looking again, it should be fine, since that script doesn't do anything related to the root directory or other paths.